aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/DeleteHandler.java

```diff
@@ -43,12 +50,11 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
         if (BooleanUtils.isTrue(request.getSnapshotRequested()) &&
                 !isReadReplica(resourceModel) &&
                 !isDBClusterMember(resourceModel)) {
-            snapshotIdentifier = generateResourceIdentifier(
-                    Optional.ofNullable(request.getStackId()).orElse(STACK_NAME),
-                    SNAPSHOT_PREFIX + Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse(RESOURCE_IDENTIFIER),
-                    request.getClientRequestToken(),
-                    SNAPSHOT_MAX_LENGTH
-            );
+            snapshotIdentifier = snapshotIdentifierFactory.newIdentifier()
+                    .withStackId(request.getStackId())
+                    .withResourceId(SNAPSHOT_PREFIX + Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse(RESOURCE_IDENTIFIER))
```

Comment:
nit:
We have set `defaultResourceId`. But we won't be able ever use the default value because we call `.withResourceId(SNAPSHOT_PREFIX + ....` and the `defaultResourceId` won't be used because we will never pass an empty resourceId.

Do you think we could replace this line with something along the lines of:
`.withResourceId(request.getLogicalResourceIdentifier() == null ? null : SNAPSHOT_PREFIX + request.getLogicalResourceIdentifier())`
This way we can actually pass a null value to trigger the use of our default value?

Reply:
Replaced with a `StringUtils.prependIfMissing(request.getLogicalResourceIdentifier(), SNAPSHOT_PREFIX)`. I aimed to avoid the ternary operator due to its ultimate clumsiness, `StringUtils` came up to the rescue as `prependIfMissing` is null-safe: it simply returns null if the first argument is also null. We will have an implicit fallback in this case.